### PR TITLE
instead of download and upload try to copy directly between containers

### PR DESCRIPTION
**Pull request by mdayican**
### Change description ###

instead of download and upload try to copy directly between containers
this may help memory usage and firewall content-type  issue

sas token required to copy
https://docs.microsoft.com/en-us/rest/api/storageservices/copy-file

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java [hunk 1]
```diff
@@ -93,27 +95,29 @@ private void handleEnvelope(
...
 
-        blobClient
-            .getBlockBlobClient()
-            .upload(
-                new ByteArrayInputStream(blobContent),
-                blobContent.length,
-                true
-            );
+        targetBlob
+            .getBlockBlobClient().copyFromUrl(sourceBlob.getBlobUrl() + "?" + createSas(sourceBlob,5));
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java]: maybe we can just set expiry locally in that method instead. if not then missing space here 😛

File: src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/BlobRejectTest.java [hunk 2]
```diff
@@ -42,6 +46,13 @@ void should_reject_crime_envelope_with_invalid_signature() throws Exception {
 
         // and
         assertFileInfoIsStored(fileName, config.crimeSourceContainer, REJECTED, true);
+
+        assertBlobIsPresentInStorage(
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/BlobRejectTest.java]: we may need to wait here for a bit as well

Comment (mdayican) [reply src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/BlobRejectTest.java]: I think first one should enough,  if it is not in the source container it should be in target

Comment (rkondratowicz) [reply src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/BlobRejectTest.java]: true!

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java [hunk 3]
```diff
@@ -93,27 +95,29 @@ private void handleEnvelope(
...
+        OffsetDateTime expiryTime =
+            OffsetDateTime.of(LocalDateTime.now().plus(5, ChronoUnit.MINUTES), ZoneOffset.UTC);
+
+        BlobServiceSasSignatureValues blobServiceSasSignatureValues =
+            new BlobServiceSasSignatureValues(
+                expiryTime,
+                new BlobContainerSasPermission().setReadPermission(true)
+            );
+
+        return sourceBlob.generateSas(blobServiceSasSignatureValues);
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java]: Sas should be for destination blob ? As you are copying to destination container ?

Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java]: sas is getting from source

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java [hunk 4]
```diff
@@ -93,27 +95,29 @@ private void handleEnvelope(
...
-            );
+        targetBlob
+            .getBlockBlobClient().copyFromUrl(sourceBlob.getBlobUrl() + "?" + createSas(sourceBlob));
+
         logger.info("File successfully uploaded to rejected container. " + loggingContext);
     }
+
+    private String createSas(BlobClient sourceBlob) {
+        OffsetDateTime expiryTime =
+            OffsetDateTime.of(LocalDateTime.now().plus(5, ChronoUnit.MINUTES), ZoneOffset.UTC);
```
Comment (mdayican) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java]: may there be time differences between aks and blob ?? , 5 minutes is  enough?

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java [hunk 5]
```diff
@@ -93,27 +95,29 @@ private void handleEnvelope(
...
 
-        blobClient
-            .getBlockBlobClient()
-            .upload(
-                new ByteArrayInputStream(blobContent),
-                blobContent.length,
-                true
-            );
+        targetBlob
+            .getBlockBlobClient().copyFromUrl(sourceBlob.getBlobUrl() + "?" + createSas(sourceBlob));
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java]: when you log you use `sourceBlob.getBlockBlobClient().getBlobUrl()` ?

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java (hunk 4 above)
Comment (nitinprabhu) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java]: both should be updated so should be fine

File: src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java (hunk 5 above)
Comment (mdayican) [reply src/main/java/uk/gov/hmcts/reform/blobrouter/services/storage/RejectedFilesHandler.java]: chnaged

**Pull request merged**